app/components/blacklight/search_context/server_item_pagination_component.rb

```diff
@@ -12,35 +12,52 @@ def initialize(search_context:, search_session:, current_document:)
       end
 
       def render?
-        @search_context.present? && (@search_context[:prev] || @search_context[:next]) && (@search_session['document_id'] == @current_document_id)
+        return false unless @search_session['document_id'] == @current_document_id
+
+        return true if total == 1
```

Comment:
I believe we need to check for `@search_context` here. Otherwise it will display on pages with no search context.